### ISSUE TITLE
easyconfig to copy lib directory

### ISSUE THREAD

**Issue by jmbuhr** [easybuilders/easybuild-easyconfigs#14044]
**tldr**: How do I tell eb / easyconfig which files / folders to copy to the install location?

Hi there, I am unable to find the easyconfig parameter to tell eb to copy the `lib` folder of a software compilation in addition to the `bin` folder. Could you point me to it?

Context: I am attempting an installation of the cp2k QM software and integrate with the MD software gromacs, hence I need not only the binary but also `libcp2k`. For this I modified https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/c/CP2K/CP2K-8.1-foss-2020b.eb with the line:
```
build_cmd_targets = ['lib2p2k', '']
```
This built both the binary and libcp2k and I am able to use cp2k after loading the module. In my `eb/software/CP2K/8.1-foss-2020b` folder I can see that the `bin/` and `data/` folders as a result of the compilation are there, but the `lib/` folder is missing.
In the log-file I can see that it did indeed built the libs in `eb/build/CP2K/8.1/foss-2020b/cp2k-8.1/lib/` but the `build` folder is cleared after successful compilation. How do I tell eb to copy the `lib` folder as well?

**Comment by jfgrimm** [comment]
Building and installing the library is already handled in the [cp2k easyblock](https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/c/cp2k.py#L95).
just adding `library = True` to the easyconfig should be sufficient

**Comment by jmbuhr** [comment]
Oh, I did not know cp2k had it's own custom easyblock, thank you! Are the easyblocks that an easyconfig file uses then simply matched by name and file class name?

**Comment by boegel** [comment]
@jmbuhr If no specific easyblock is specified, then a custom easyblock is assumed, and that's indeed base on the software name; see also https://docs.easybuild.io/en/latest/Writing_easyconfig_files.html#easyblock-specification

**Comment by jmbuhr** [comment]
Thank you very much :) If I may, I do have a follow-up question: Setting `library = True` did indeed build and copy some of the libraries, but not all the files are there and the build fails with `Sanity check failed: no file found at 'lib/libcp2k.a' in <...>/eb/software/CP2K/8.1-foss-2020b`
![image](https://user-images.githubusercontent.com/17450586/134775154-e273c1a2-7b66-416a-bb46-c933c2fe921b.png)
there are a lot of lib-files in there, but that one is missing (and I think it is important).

**Comment by boegel** [comment]
@jmbuhr Any trace of compilation errors in the build log when `libcp2k.a` is being built? Can you share the EasyBuild log file?

**Comment by jmbuhr** [comment]
No mention of other errors apart from the one where it fails the test:
[easybuild-CP2K-8.1-20210924.230123.HDXST.log](https://github.com/easybuilders/easybuild-easyconfigs/files/7229996/easybuild-CP2K-8.1-20210924.230123.HDXST.log)

**Comment by jmbuhr** [comment]
I was able to build it by changing the EasyBlock: According to the installation instructions, the library is built by first doing a regular built followed by one with make `libcp2k` (https://github.com/cp2k/cp2k/blob/master/INSTALL.md#3c-building-cp2k-as-a-library and https://gitlab.com/gromacs/gromacs/-/blob/2021.2-qmmm/INSTALL-dev).

If I replace:
https://github.com/easybuilders/easybuild-easyblocks/blob/74a385d64744838c86c6b4b107cdc03c097b0f20/easybuild/easyblocks/c/cp2k.py#L670-L673
```
        if self.cfg['library']:
            cmd += ' libcp2k'
        run_cmd(cmd + " all", log_all=True, simple=True, log_output=True)
```

with

```
        run_cmd(cmd, log_all=True, simple=True, log_output=True)

        if self.cfg['library']:
            cmd += ' libcp2k'
            run_cmd(cmd, log_all=True, simple=True, log_output=True)
```

it works as expected. Should I make a PR for the EasyBlock?

**Comment by boegel** [comment]
@jmbuhr Oh, maybe they changed this in a recent CP2K version then?

A pull request for the CP2K easyblock would be great!

**Comment by jmbuhr** [comment]
I will try to find out of this is the case and add something along the lines of `if LooseVersion(self.version) >= ...`

**Comment by boegel** [comment]
@jmbuhr If the approach with two separate commands also works for older CP2K versions, there's no need for the version check.
